keep/providers/mongodb_provider/mongodb_provider.py

```diff
@@ -0,0 +1,126 @@
+    )
+    username: str = dataclasses.field(
+        metadata={"required": False, "description": "MongoDB username"}
+    )
+    password: str = dataclasses.field(
+        metadata={"required": False, "description": "MongoDB password", "sensitive": True}
+    )
+    host: str = dataclasses.field(
+        metadata={"required": False, "description": "MongoDB hostname"}
+    )
+    database: str = dataclasses.field(
+        metadata={"required": False, "description": "MongoDB database name"}
+    )
+
+class MongoDBProvider(BaseProvider):
+    """Enrich alerts with data from MongoDB."""
+
+    PROVIDER_SCOPES = [
+        ProviderScope(
+            name="connect_to_server",
+            description="The user can connect to the server",
+            mandatory=True,
+            alias="Connect to the server",
+        )
+    ]
+
+    def __init__(
+        self, context_manager: ContextManager, provider_id: str, config: ProviderConfig
+    ):
+        super().__init__(context_manager, provider_id, config)
+        self.client = None
+
+    def validate_scopes(self):
+        """
+        Validates that the user has the required scopes to use the provider.
+        """
+        try:
+            client = self.__generate_client()
+            client.close()
+            scopes = {
+                "connect_to_server": True,
+            }
+        except Exception as e:
+            self.logger.exception("Error validating scopes")
+            scopes = {
+                "connect_to_server": str(e),
+            }
+        return scopes
+
+    def __generate_client(self):
+        """
+        Generates a MongoDB client.
+
+        Returns:
+            pymongo.MongoClient: MongoDB Client
+        """
+        if self.authentication_config.uri:
+            client = MongoClient(self.authentication_config.uri)
+        else:
+            client = MongoClient(
+                f"mongodb://{self.authentication_config.username}:{self.authentication_config.password}@{self.authentication_config.host}/{self.authentication_config.database}"
+            )
+        return client
+
+    def dispose(self):
+        try:
+            self.client.close()
+        except Exception:
+            self.logger.exception("Error closing MongoDB connection")
+
+    def validate_config(self):
+        """
+        Validates required configuration for MongoDB's provider.
+        """
+        self.authentication_config = MongoDBProviderAuthConfig(
+            **self.config.authentication
+        )
+
+    def _query(
+        self, query="", as_dict=False, single_row=False, **kwargs: dict
+    ) -> list | tuple:
+        """
+        Executes a query against the MongoDB database.
+
+        Returns:
+            list | tuple: list of results or single result if single_row is True
+        """
+        client = self.__generate_client()
+        database = client[self.authentication_config.database]
+        results = list(database.eval(query))
```

Comment:
https://www.mongodb.com/docs/v4.2/reference/method/db.eval/ Eval is not supported on latest mongodb versions?